packages/web-components/fast-foundation/src/button/button.ts

```diff
@@ -157,8 +157,40 @@ export class Button extends FormAssociatedButton {
 
         this.proxy.setAttribute("type", this.type);
         this.handleUnsupportedDelegatesFocus();
+
+        const elements = this.shadowRoot?.querySelectorAll("span");
+        if (elements) {
+            const spans: HTMLSpanElement[] = Array.from(elements);
+            spans.forEach((span: HTMLSpanElement) => {
+                span.addEventListener("click", this.handleClick);
+            });
+        }
+    }
+
+    /**
+     * @internal
+     */
+    public disconnectedCallback(): void {
+        super.disconnectedCallback();
+        const elements = this.shadowRoot?.querySelectorAll("span");
```

Comment:
We spoke offline but I wanted to make the note; We shouldn't access `this.shadowRoot` here because it will be `null` in cases of closed shadow roots and then this won't exhibit the desired behavior.

Reply:
I've made the update to add the click handler on the spans themselves @nicholasrice please take a look when you get a chance (particularly in the start-end.ts file), thanks!

Reply:
Looks good, thanks!